vendors/espressif/esp-idf/components/driver/rtc_module.c

```diff
@@ -995,7 +995,7 @@ esp_err_t touch_pad_filter_start(uint32_t filter_period_ms)
     }
     if (s_touch_pad_filter->timer == NULL) {
         s_touch_pad_filter->timer = xTimerCreate("filter_tmr", filter_period_ms / portTICK_PERIOD_MS, pdFALSE,
-        NULL, touch_pad_filter_cb);
+        NULL, (void(*)(TimerHandle_t))touch_pad_filter_cb);
```

Comment:
It is generally better to let the compiler check the prototype, if we need a cast here there is probably something wrong with the prototype. What was the warning here?

I would really like to avoid casting a function pointer at all cost!

Reply:
The warning here was `warning: passing argument 5 of 'xTimerCreate' from incompatible pointer type [-Wincompatible-pointer-types]`. This is because `TimerHandle_t` is declared as `typedef struct tmrTimerControl * TimerHandle_t;` in AmazonFreeRTOS, while it is declared as `typedef void * TimerHandle_t;` in esp-idf which results in a warning.

Reply:
@cobusve To keep compatibility of version of FreeRTOS used in ESP-IDF this change is required (https://github.com/espressif/esp-idf/blob/release/v3.3/components/freertos/include/freertos/timers.h#L118). This is approved change and will reflect in public ESP-IDF github repository soon.

Reply:
Ok Mahavir, with your assurance that this is not diverging from the esp-idf release I can approve it for you.